components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1251,6 +1251,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 context = prebuiltWorkspace;
             }
 
+            await mayStartWorkspacePromise;
```

Comment:
I'm guessing writing a test for this would be too complicated given we've got nothing in place currently?

Reply:
It would be quite an effort as the downstream components would all need to be available or mocked out. Given that we hopefully can soon implement a workspace service based on grpc API. I believe that to be a better moment to writing a test infrastructure.